### ISSUE TITLE
Tensorboard not fully functional behind nginx proxy

### ISSUE THREAD

**Issue by bajo** [tensorflow/tensorflow#9614]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
Yes. The ngnix configuration to proxy tensorboard from port 6006 to port 80. Details below.
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**:
Source
- **TensorFlow version (use command below)**:
('v1.0.1-2-g250e72c-dirty', '1.0.1')
- **Bazel version (if compiling from source)**:
Build label: 0.4.5
Build target: bazel-out/local-fastbuild/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Thu Mar 16 12:19:38 2017 (1489666778)
Build timestamp: 1489666778
Build timestamp as int: 1489666778
- **CUDA/cuDNN version**:
8.0.61/cudnn-8.0-linux-x64-v5.1
- **GPU model and memory**:
"TITAN X (Pascal) and 12187 MBytes
- **Exact command to reproduce**:
tensorboard  --logdir /tmp/retrain_logs/

### Describe the problem
I follow the image retraining tutorial  [https://www.tensorflow.org/tutorials/image_retraining]. As we have different applications providing a webinterface and a firewall that blocks all ports except 22(ssh) and 80(http)/443(https) we want to put all of them behind a nginx proxy. 
With the config below the tensorboard shows up, however no data is displayed.
When disabling all other applications / services that run on port 80 and forcing tensorboard to deliver on port 80 with `sudo tensorboard --port 80 --logdir /tmp/retrain_logs` the data graphs (ie. accuracy_1, cross_entropy_1, etc.) are displayed. 

### Source code / logs
`server {
        listen 80 default_server;
        listen [::]:80 default_server;

        root /var/www/html;

        # Add index.php to the list if you are using PHP
        index index.html index.htm index.nginx-debian.html;

        server_name _;

        location / {
                # First attempt to serve request as file, then
                # as directory, then fall back to displaying a 404.
                try_files $uri $uri/ =404;
        }
        location /tensorflow {
                rewrite ^/tensorflow(.*) /$1 break;
                proxy_pass http://127.0.0.1:6006/;
        }
}
`

**Comment by jart** [comment]
It's worked for me in the past when I configured it as follows:

```nginx
server {
  listen 80 default_server;
  listen [::]:80 default_server ipv6only=on;

  location / {
    proxy_http_version 1.1;
    proxy_pass http://127.0.0.1:6006;
  }
}
```

Since you're chrooting the app, you need to take into consideration that asset and XHR requests get sent to a whole bunch of other endpoints (e.g. /data, /runs, etc.) using absolute paths.

Refactoring TensorBoard to make all asset requests relative, and then maybe rooting XHRs under a single prefix like `/_/tensorboard/` would be nice (CC: @dandelionmane) but also a nontrivial undertaking.

I would recommend creating a separate hostname for TensorBoard. For example, say `http://tensorboard.you.com` rather than `http://you.com/tensorboard`. Then configure a virtual host in nginx.

You wouldn't by any chance be building a productionized service for TensorBoard? Because if that's something you're interested in having, we should talk more.

**Comment by ml9951** [comment]
@jart, your nginx configuration worked for me.

**Comment by dandelionmane** [comment]
Closing since it sounds like Justine's solution works.

**Comment by jlzhao27** [comment]
@decentralion I'd like to re-open this issue since we have a use case where it is not possible to deploy tensorboard on `/`. 

We are deploying several services behind a single DNS name due to certificate signing and other restrictions. We want to be able to add tensorboard on this proxy at some sub-path like `<my-host>/tensorboard` but the static paths break this. Seems like many other apps already support this such as Jupyter's baseUrl: https://jupyter-notebook.readthedocs.io/en/stable/config.html

**Comment by ogre0403** [comment]
@decentralion  we also have similar issue since we try to setup a tensorflow lab environment on kubernetes. 
Our use case is use kubernetes Ingress for user to access jupyter/tensorboard from a url having sub-path.  For Jupyter, we use baseUrl mentioned by @jlzhao27 to do this, but we can't do this for tensorboard.

**Comment by mabrowning** [comment]
Since this is a top Google result for "tensorboard reverse proxy", let it be known that tensorboard now has a `--path_prefix=/whatever/you/want/tensorboard` argument for serving at a non-root URL.

**Comment by bkgoodman** [comment]
Also important to note - that I was using apache reverse  proxy to send `/tensorboard` to the (tensorboard at port) 6006. I passed `--path_prefix=/tensorboard` to tensorboard to run it, but I ultimatley had to connect to `http://1.2.3.4/tensorboard/tensorboard/` to actually get it to work. (Notice that "tensorboard" is specified *twice* and there is a trailing slash on the end of the URL!)

My Apache config was as folllows:

```
ProxyPass "/tensorboard"  "http://127.0.0.1:6006/"
ProxyPassReverse "/tensorboard"  "http://127.0.0.1:6006/"
```
